src/basic/log.c

```diff
@@ -552,7 +552,7 @@ static int write_to_kmsg(
         if (kmsg_fd < 0)
                 return 0;
 
-        if (!ratelimit_below(&ratelimit))
+        if (log_get_max_level() != LOG_DEBUG && !ratelimit_below(&ratelimit))
                 return 0;
```

Comment:
hmm, so we added this for political reasons: the generic kernel cmdline param "debug" will enable debug logging for both kernel and various userspace bits, including all of systemd. Some kernel people hated that, they wanted kmsg for themselves, and really didn't care that early userspace doesn't really have any other place to log to.

So we added the ratelimiting here, so that we don't spill too much into kmsg, but still get some useful info in there. It's an exercise in not flooding kmsg and not giving the kernel folks too much of a reason to be pissed.

Hence, I am not sure we should drop the ratelimiting here on debug, because it's precisely what we added this for. I'd be fine with raising the limits though.

Reply:
we could also scale the ratelimit via what syslog(SYSLOG_ACTION_SIZE_BUFFER) returns. allow more messages per time window when that value is large, and less when it is small. That way, if people specify log_buf_len=2M we will automatucally log more too.

Reply:
I just bumped the limit. I'm not entirely sure what the right scaling factor would be when using the kernel log buffer size.